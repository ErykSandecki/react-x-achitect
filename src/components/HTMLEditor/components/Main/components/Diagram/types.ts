// types
import { BlockType, PortType, T2DCoordinates, TBlock, TConnector } from 'types';

export type TDraggableConnectorSource = {
  blockId: string;
  blockType: BlockType;
  portType: PortType;
};

export type TDraggableConnectorTarget = {
  blockId: string;
  blockType: BlockType;
  portType: PortType;
};

export type TDraggableConnector = {
  showIndicator: boolean;
  source?: TDraggableConnectorSource;
  sourceCoordinates: T2DCoordinates;
  target?: TDraggableConnectorTarget;
  targetCoordinates: T2DCoordinates;
};

export type TPossiblePort = T2DCoordinates & {
  blockId: string;
  blockType: BlockType;
  height: number;
  portType: PortType;
  width: number;
};

export type TShadowConnector = Pick<
  TConnector,
  'joinOperator' | 'relatedWithBlockId' | 'towardsTo'
> & {
  sourceBlock: Partial<TBlock> & { portType: PortType };
  targetBlock: Partial<TBlock> & { portType: PortType };
};
