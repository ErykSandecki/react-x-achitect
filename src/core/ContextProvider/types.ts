// types
import { Theme } from 'types/enums';

export type TContext = {
  scrollLock: boolean;
  setScrollLock: (scrollLock: boolean) => void;
  setTheme: (theme: Theme) => void;
  theme: Theme;
};

export type TActionOnChangeTheme = (payload: Theme) => {
  payload;
  type: string;
};
