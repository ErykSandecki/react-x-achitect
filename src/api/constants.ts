// types
import { Header } from './enums';

export const BASE_URL = '';

export const HEADERS = {
  [Header.contentType]: 'application/json',
};

export const RESPONSE_DEV_TIMEOUT = 'dev-response-timeout';

export const translationNameSpace = 'response';
