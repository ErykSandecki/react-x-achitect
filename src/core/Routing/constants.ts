export const translationNameSpace = 'routing';
export const guardNameSpace = `${translationNameSpace}.guard`;
