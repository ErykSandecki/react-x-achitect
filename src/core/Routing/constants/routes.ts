export enum RouteName {
  home = 'home',
}

export const ROUTES = {
  [RouteName.home]: '/',
};
