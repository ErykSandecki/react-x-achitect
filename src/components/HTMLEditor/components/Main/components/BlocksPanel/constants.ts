// others
import { translationNameSpace as parentNameSpace } from '../../../../constants';

export const BLOCKS_PANEL_ID = 'blocksPanel';
export const MIN_HEIGHT = 283;

export const translationNameSpace = `${parentNameSpace}.blocksPanel`;
