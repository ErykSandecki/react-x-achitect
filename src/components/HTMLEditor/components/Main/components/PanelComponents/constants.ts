// others
import { translationNameSpace as parentNameSpace } from '../../../../constants';

export const BLOCKS_PANEL_ID = 'panelComponents';
export const MIN_HEIGHT = 283;

export const translationNameSpace = `${parentNameSpace}.panelComponents`;
