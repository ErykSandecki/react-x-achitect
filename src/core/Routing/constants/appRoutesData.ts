// components
import HomePage from 'pages/HomePage/HomePage';

// types
import { RouteName } from './routes';
import { TAppRoutesData } from '../types';

export const APP_ROUTES_DATA: TAppRoutesData = [
  {
    Component: HomePage,
    guards: [],
    name: RouteName.home,
  },
];
