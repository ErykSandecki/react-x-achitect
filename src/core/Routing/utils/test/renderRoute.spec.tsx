import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

// components
import HomePage from 'pages/HomePage/HomePage';

// core
import { renderRoute } from '../renderRoute';

// mocks
import { routerStateMock } from 'test/mocks/reducer/routerMock';

// store
import { configureStore } from 'store/store';

// types
import { RouteName } from '../../constants/routes';
import { TAppRouteData } from '../../types';

// utils
const stateMock = {
  ...routerStateMock,
};

describe('renderRoute snaphots', () => {
  const store = configureStore(stateMock);

  it('should render renderRoute', () => {
    // mock
    const routeData: TAppRouteData = {
      Component: HomePage,
      name: RouteName.home,
    };

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>{renderRoute(routeData)}</BrowserRouter>
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render when routeDataArg is string', () => {
    // mock
    const routeData: TAppRouteData['name'] = RouteName.home;

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>{renderRoute(routeData)}</BrowserRouter>
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
