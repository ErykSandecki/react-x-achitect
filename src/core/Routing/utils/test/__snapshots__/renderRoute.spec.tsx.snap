// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderRoute snaphots should render renderRoute 1`] = `
<DocumentFragment>
  <div
    class="Page Page--light"
  >
    <div
      class="ErrorPage ErrorPage--light"
    >
      <div
        class="ErrorPage__background ErrorPage__background--light"
      />
      <div
        class="ErrorPage__content ErrorPage__content--light"
      >
        <div
          class="ErrorPage__logo ErrorPage__logo--light"
          data-test-logo=""
        />
        <h3
          class="ErrorPage__title ErrorPage__title--light typography_Typography__Hvofe typography_Typography--light__ma6Vx typography_Typography--regular__G4QAa typography_Typography--normal__2w6Rv typography_Typography--h3__yKIyB typography_Typography--without-margin__uJld2"
          data-test-text=""
          style="text-align: inherit;"
        >
          Interval Server Error
        </h3>
        <p
          class="ErrorPage__description ErrorPage__description--light typography_Typography__Hvofe typography_Typography--light__ma6Vx typography_Typography--regular__G4QAa typography_Typography--normal__2w6Rv typography_Typography--p__efCcF typography_Typography--without-margin__uJld2"
          data-test-text=""
          style="text-align: inherit;"
        >
          The server has been deserted for a while. Please be patient or try again later.
        </p>
        <div />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renderRoute snaphots should render when routeDataArg is string 1`] = `<DocumentFragment />`;
